MarsRover.Tests/ControllerShould.cs

```diff
@@ -6,67 +6,84 @@ namespace MarsRover.Tests
 {
     public class ControllerShould
     {
+        Controller _controller;
+        Mock<IInput> _mockInput;
+        StubOutput _output;
+        FileMapInput _fileMapInput;
+
+        public ControllerShould()
+        {
+            _mockInput = new Mock<IInput>();
+            _output = new StubOutput();
+            _fileMapInput = new FileMapInput();
+            _controller = new Controller(_mockInput.Object, _output, _fileMapInput);
+        }
         [Fact]
         public void Run_ReturnsMapOutput_GivenRoverCommandsThatResultInNoObstacle()
         {
-            var mockInput = new Mock<IInput>();
-            mockInput.SetupSequence(_ => _.ReadLine())
+            _mockInput.SetupSequence(_ => _.ReadLine())

```

Comment:
probably worth using something like `i` for input or an arbitrary character other than a discard (`_`) which implies that it won't be used